cache/cache.go

```diff
@@ -0,0 +1,190 @@
+package cache
+
+import (
+	"container/list"
+	"sync"
+)
+
+// Cache is a bounded-size in-memory cache of sized items with a configurable eviction policy
+type Cache interface {
+	// Get retrieves items from the cache by key.
+	// If an item for a particular key is not found, its position in the result will be nil.
+	Get(keys ...string) []Item
```

Comment:
If the consumer requires a list of items, this means you only have to lock the mutex once for a large range.  This is usually a lot more performant than many locks back to back.  Also ensures atomicity.  

I've found at least one large advantage to the ordering.  Let's say I expect something to be cached and I've passed in a list of keys.  When I do nil detection on the other side and find the nth item is nil, it's trivial for me to log the associated key.